### PR TITLE
added pagination with tests

### PR DESCRIPTION
**Pull request by plokijro**
https://trello.com/c/cylpv3kJ/42-be-add-pagination

### CODE REVIEW

File: app/controllers/api/v1/recipes_controller.rb [hunk 1]
```diff
@@ -4,7 +4,17 @@ class RecipesController < ApplicationController
...
+        recipes = Recipe.all.page(params[:page]).per(params[:per_page])
+
+        render json:
+        {
+          data: ActiveModel::SerializableResource.new(recipes, each_serializer: RecipeIndexSerializer),
+          pagination: {
+            page: params[:page].presence || recipes.current_page,
+            per_page: params[:per_page].presence || recipes.default_per_page,
+            total: recipes.total_pages
+          }
```
Comment (PatrykTyszka) [inline app/controllers/api/v1/recipes_controller.rb:16]:
```suggestion
            recipes: ActiveModel::SerializableResource.new(recipes, each_serializer: RecipeIndexSerializer),
            page: params[:page].presence || recipes.current_page,
            per_page: params[:per_page].presence || recipes.default_per_page,
            total: recipes.total_pages
          }
```

File: spec/controllers/api/v1/recipes_controller_spec.rb [hunk 2]
```diff
@@ -3,12 +3,27 @@
...
+
+    it 'returns two recipes for second page' do
+      get :index, params: { per_page: 2, page: 2 }
+      expect(JSON.parse(response.body)['data'].size).to eq(2)
+    end
+
+    it 'returns two recipes for first page' do
+      get :index, params: { per_page: 2 }
+      expect(JSON.parse(response.body)['data'].size).to eq(2)
+    end
```
Comment (PatrykTyszka) [inline spec/controllers/api/v1/recipes_controller_spec.rb:22]: ... so these two tests? IMO checks the same case.

Comment (plokijro) [reply spec/controllers/api/v1/recipes_controller_spec.rb:22]: In this second test, I wanted to show that without a page in params, it will work anyway

Comment (PatrykTyszka) [reply spec/controllers/api/v1/recipes_controller_spec.rb:22]: It would be better to show up that we have different objects for each page.

**Review by PatrykTyszka — APPROVED** [review]

**Pull request merged**